collect_app/src/main/java/org/odk/collect/android/google/DriveHelper.java

```diff
@@ -0,0 +1,248 @@
+     * @param destinationFolderID Id of the folder into which the file has to be uploaded
+     * @param toUpload            The file which is to be uploaded
+     * @return id of the file if uploaded successfully
+     */
+    public String uploadFileToDrive(String mediaName, String destinationFolderID, File toUpload)
+            throws IOException {
+
+        //adding meta-data to the file
+        com.google.api.services.drive.model.File fileMetadata =
+                new com.google.api.services.drive.model.File()
+                        .setName(mediaName)
+                        .setViewersCanCopyContent(true)
+                        .setParents(Collections.singletonList(destinationFolderID));
+
+        String type = FileUtils.getMimeType(toUpload.getPath());
+        FileContent mediaContent = new FileContent(type, toUpload);
+
+        return drive.files()
+                .create(fileMetadata, mediaContent)
+                .setFields("id, parents")
+                .setIgnoreDefaultVisibility(true)
+                .execute()
+                .getId();
+    }
+
+    /**
+     * Creates a new folder in google drive
+     *
+     * @param folderName The name of the new folder
+     * @param parentId   The id of the folder in which we want to create the new folder
+     * @return the folder object if created successfully
+     */
+    private com.google.api.services.drive.model.File createFolderInDrive(@NonNull String folderName,
+                                                                         @Nullable String parentId)
+            throws IOException {
+
+        //creating a new folder
+        com.google.api.services.drive.model.File fileMetadata = new
+                com.google.api.services.drive.model.File();
+        fileMetadata.setName(folderName);
+        fileMetadata.setMimeType("application/vnd.google-apps.folder");
+        if (parentId != null) {
+            fileMetadata.setParents(Collections.singletonList(parentId));
+        }
+
+        com.google.api.services.drive.model.File folder;
+        folder = drive.files().create(fileMetadata)
+                .setFields("id")
+                .execute();
+
+        //adding the permissions to folder
+        Permission sharePermission = new Permission()
+                .setType("anyone")
+                .setRole("reader");
+
+        drive.permissions().create(folder.getId(), sharePermission)
+                .setFields("id")
+                .execute();
+
+        return folder;
+    }
+
+    /**
+     * Fetches the list of files from google drive for a given folder.
+     *
+     * @param folderName (optional) The name of folder whose files are to be fetched
+     *                   If folderName is null then all files are fetched from the drive
+     * @param parentId   (optional) The id of the parent folder containing the given folder
+     */
+    public List<com.google.api.services.drive.model.File> getFilesFromDrive(
```

Comment:
Same as above, we should probably break out the request string creation and the actual Drive request. That way we can test them separately.